digicampipe/image/sky_image.py

```diff
@@ -0,0 +1,555 @@
+                print('Retrieving file from', url, 'to', fn)
+                f = None
+                while f is None:
+                    try:
+                        f = urlopen(url)
+                    except HTTPError as err:
+                        print("ERROR", err, "getting url", url)
+                        f = None
+                        break
+                if f is None:
+                    continue
+                txt = f.read()
+                w = open(fn, 'wb')
+                w.write(txt)
+                w.close()
+                print('Wrote to', fn)
+            # coordinate system
+            self.wcs = None
+            self.reference_pixel = None
+            self.reference_ra_dec = None
+            self.cd_matrix = None
+            if success:
+                try:
+                    header_wcs = fits.open(wcs_file)[0].header
+                    self.wcs = WCS(header_wcs)
+                    # reference point
+                    self.reference_pixel = (header_wcs['CRPIX1'], header_wcs['CRPIX2'])
+                    self.reference_ra_dec = (header_wcs['CRVAL1'], header_wcs['CRVAL2'])
+                    # transformation matrix
+                    self.cd_matrix = np.array(((header_wcs['CD1_1'], header_wcs['CD1_2']),
+                                               (header_wcs['CD2_1'], header_wcs['CD2_2'])))
+                except FileNotFoundError:
+                    pass
+            # stars position in image
+            self.stars_pixel = None
+            if success:
+                try:
+                    stars_data = fits.open(corr_file)[1].data
+                    self.stars_pixel = np.array((stars_data['index_x'], stars_data['index_y']))
+                except FileNotFoundError:
+                    pass
+
+
+class LidCCDImage(object):
+    """
+    LidCCDImage takes an image and  a list of areas of interest and create a SkyImage for each area.
+    """
+    def __init__(self, filename, crop_pixels1, crop_pixels2, image_static=None,
+                 threshold=None, scale_low_images_deg=None, scale_high_images_deg=None,
+                 guess_ra_dec=None, guess_radius=None):
+        if type(filename) is not str:
+            raise AttributeError('filename must be a string.')
+        self.filename = filename
+        if type(crop_pixels1) is not list:
+            raise AttributeError('crop_pixels1 must be a list of pixels.')
+        if type(crop_pixels2) is not list:
+            raise AttributeError('crop_pixels2 must be a list of pixels.')
+        if len(crop_pixels1) != len(crop_pixels2):
+            raise AttributeError('crop_pixels1 and crop_pixels2 lists must have the same length.')
+        image = fits.open(self.filename)[0].data
+        self.image_shape = image.shape
+        self.crop_pixels1 = []
+        self.crop_pixels2 = []
+        self.sky_images = []
+        self.sky_images_shape = []
+        print('divide', filename, 'in', len(crop_pixels1), 'sub-areas')
```

Comment:
this should maybe become a log-statement

Reply:
I'm not so sure how to do that :-/